### PR TITLE
[BugFix] Fix timeout  when getting kafka partition offset

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by chaoyli — DISMISSED** [review]

**Review by chaoyli — DISMISSED** [review]

File: be/src/runtime/routine_load/routine_load_task_executor.cpp [hunk 1]
```diff
@@ -115,12 +116,25 @@ Status RoutineLoadTaskExecutor::get_kafka_partition_offset(const PKafkaOffsetPro
...
+    auto it = ctx.kafka_info->properties.find("group.id");
+    if (it == ctx.kafka_info->properties.end()) {
+        group_id = "unknown";
+    } else {
+        group_id = it->second;
+    }
+    LOG(INFO) << "group id " << group_id << " get kafka partition offset. used time(ms): "
+              << std::chrono::duration_cast<std::chrono::microseconds>(end - start).count() / 1000;
+    if (!st.ok()) {
+        LOG(INFO) << "group id " << group_id << " get kafka partition offset. error: " << st.to_string();
```
Comment (meegoo) [inline be/src/runtime/routine_load/routine_load_task_executor.cpp:136]: WARNING LOG?

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 2]
```diff
@@ -290,11 +290,17 @@ Status KafkaDataConsumer::get_partition_offset(std::vector<int32_t>* partition_i
...
     beginning_offsets->reserve(partition_ids->size());
     latest_offsets->reserve(partition_ids->size());
+    MonotonicStopWatch watch;
+    watch.start();
     for (auto p_id : *partition_ids) {
         int64_t beginning_offset;
         int64_t latest_offset;
+        auto left_ms = timeout - watch.elapsed_time() / 1000 / 1000;
+        if (left_ms <= 0) {
+            return Status::TimedOut("get kafka info timeout").to_protobuf(response->mutable_status());
```
Comment (imay) [inline be/src/runtime/routine_load/data_consumer.cpp:300]: Please write this complex expression in multiple lines.

File: be/src/runtime/routine_load/routine_load_task_executor.h [hunk 3]
```diff
@@ -89,10 +89,10 @@ class RoutineLoadTaskExecutor {
     Status submit_task(const TRoutineLoadTask& task);
 
     Status get_kafka_partition_meta(const PKafkaMetaProxyRequest& request, std::vector<int32_t>* partition_ids,
-                                    int timeout);
+                                    int timeout, std::string& group_id);
```
Comment (meegoo) [inline be/src/runtime/routine_load/routine_load_task_executor.h:92]: const std::string& group_id

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 4]
```diff
@@ -290,11 +290,17 @@ Status KafkaDataConsumer::get_partition_offset(std::vector<int32_t>* partition_i
...
     beginning_offsets->reserve(partition_ids->size());
     latest_offsets->reserve(partition_ids->size());
+    MonotonicStopWatch watch;
+    watch.start();
     for (auto p_id : *partition_ids) {
         int64_t beginning_offset;
         int64_t latest_offset;
+        auto left_ms = timeout - watch.elapsed_time() / 1000 / 1000;
+        if (left_ms <= 0) {
+            return Status::TimedOut("get kafka info timeout");
```
Comment (rickif) [inline be/src/runtime/routine_load/data_consumer.cpp:300]:
```suggestion
            return Status::TimedOut("get kafka partition offset timeout");
```

Comment (rickif) [inline be/src/runtime/routine_load/data_consumer.cpp:300]:
```suggestion
            return Status::TimedOut("get kafka partition offset timeout");
```

File: be/src/runtime/routine_load/routine_load_task_executor.cpp [hunk 5]
```diff
@@ -43,12 +43,14 @@
...
 #include "util/defer_op.h"
+#include "util/stopwatch.hpp"
 #include "util/uid_util.h"
 
 namespace starrocks {
 
 Status RoutineLoadTaskExecutor::get_kafka_partition_meta(const PKafkaMetaProxyRequest& request,
-                                                         std::vector<int32_t>* partition_ids, int timeout_ms) {
+                                                         std::vector<int32_t>* partition_ids, int timeout_ms,
+                                                         std::string& group_id) {
```
Comment (rickif) [inline be/src/runtime/routine_load/routine_load_task_executor.cpp:53]:
```suggestion
                                                         std::string* group_id) {
```

Comment (rickif) [reply be/src/runtime/routine_load/routine_load_task_executor.cpp:53]: The variable should be set via a pointer, not a reference.

**Review by chaoyli — APPROVED** [review]

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 6]
```diff
@@ -290,11 +290,17 @@ Status KafkaDataConsumer::get_partition_offset(std::vector<int32_t>* partition_i
     _last_visit_time = time(nullptr);
     beginning_offsets->reserve(partition_ids->size());
     latest_offsets->reserve(partition_ids->size());
+    MonotonicStopWatch watch;
+    watch.start();
     for (auto p_id : *partition_ids) {
         int64_t beginning_offset;
         int64_t latest_offset;
+        auto left_ms = timeout - watch.elapsed_time() / 1000 / 1000;
+        if (left_ms <= 0) {
```
Comment (rickif) [inline be/src/runtime/routine_load/data_consumer.cpp:299]: How about adding a LOG(WARNING) here?

Comment (zaorangyang) [reply be/src/runtime/routine_load/data_consumer.cpp:299]: > How about adding a LOG(WARNING) here?

If this fails, the upper caller will print a warning log

**Review by meegoo — APPROVED** [review]

**Pull request merged**